app/Http/Controllers/admin/StatisticsController.php

```diff
@@ -1,114 +1,215 @@
+                    'statisticalDataFetchMethod' => 'fetchSalesStatisticalData'
+                ],
+        ];
+    
+        if (!array_key_exists($selectedTableId, $tableIdMethodMap)) {
+            return response()->json(['error' => 'Invalid table ID']);
+        }
+    
+        $yearlyDataFetchMethod = $tableIdMethodMap[$selectedTableId]['yearlyDataFetchMethod'];
+        $table = $tableIdMethodMap[$selectedTableId]['table'];
+        $statisticalDataFetchMethod = $tableIdMethodMap[$selectedTableId]['statisticalDataFetchMethod'];
 
-                // Store the result in the results array
-                $numericalDataNumByAttribute[$attribute->name][$monthName] = $count;
+        $attributes = $this->getAttributes($table);
+        $yearlyData = $this->$yearlyDataFetchMethod($selectedYear);
+        $statisticalData = $this->$statisticalDataFetchMethod($attributes, $months, $yearlyData, $selectedTableId);
+    
+        // Add "Total" attribute
+        $statisticalData['Total'] = [];
+        foreach($months as $monthName) {
+            $total = 0;
+            foreach($attributes as $attribute) {
+                $total += $statisticalData[$attribute->name][$monthName];
             }
+            $statisticalData['Total'][$monthName] = $total;
         }
-
+    
         // Convert the results array to an object
         $fetchedData = [
-            'year' => $selectedYear,
             'months' => array_values($months),
-            'attributes' => $attributes->pluck('name')->toArray(),
-            'numericalDataNumByAttribute' => $numericalDataNumByAttribute,
+            'attributes' => array_merge($attributes->pluck('name')->toArray(), ['Total']),
+            'statisticalData' => $statisticalData,
         ];
-
         return response()->json($fetchedData);
     }
-    */
     
-    /**
-     * Show the statistics page.
-     *
-     * This method fetches the registration data by default and passes it to the 'admin.statistics.show' view.
-     */
-    public function showStatisticsTest(){
-        $defaultData = getDefaultData();
-        return view('admin.statistics.show',['data' => $defaultData]);
+    private function getAttributes($table) {
+        if($table === 'users'){
+            $attributes = $this->attribute->whereIn('id', function($query) {
+                $query->select('attribute_id')->from($this->user->getTable())->distinct();
+            })->get();
+        } else if ($table === 'reservations'){
+            $attributes = $this->attribute->whereIn('id', function($query) {
+                $query->select('attribute_id')
+                    ->from($this->area->getTable())
+                    ->whereIn('id', function($subQuery) {
+                        $subQuery->select('area_id')->from($this->reservation->getTable())->distinct();
+                    })
+                    ->distinct();
+            })->get();
+        }
+        return $attributes;
     }
```

Comment:
declare $attributes = array(); This for cases when nothing matched for if ane else if

Reply:
Thank you for your suggestion.
Added $attributes = array();
https://github.com/kredo-school/19tha_easepark/blob/3ae8512fa1fdcbd8c40ef0e8474b44eb1b05ca15/app/Http/Controllers/admin/StatisticsController.php#L97-L114